src/SprykerEco/Zed/Unzer/Business/Checkout/Mapper/UnzerCheckoutPostSaveMapper.php

```diff
@@ -112,14 +112,14 @@ protected function mapQuoteItemTransferToUnzerBasketItemTransfer(
     ): UnzerBasketItemTransfer {
         return $unzerBasketItemTransfer
             ->setBasketItemReferenceId(
-                $this->utilTextService->generateUniqueId($itemTransfer->getSku(), true),
+                $this->utilTextService->generateUniqueId($itemTransfer->getSkuOrFail(), true),
             )
             ->setQuantity($itemTransfer->getQuantity())
-            ->setAmountGross($itemTransfer->getSumGrossPrice() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
-            ->setAmountVat($itemTransfer->getSumTaxAmount() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
-            ->setAmountDiscount($itemTransfer->getSumDiscountAmountAggregation() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
-            ->setAmountPerUnit($itemTransfer->getUnitPriceToPayAggregation() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
-            ->setAmountNet($itemTransfer->getSumNetPrice() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
+            ->setAmountGross((int)$itemTransfer->getSumGrossPrice() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
```

Comment:
same here. Unzer transfer properties should be float

Reply:
Method getGrandTotal() returns int|null ... If you don't cast return value getGrandTotal() to integer you would get Left operand cannot be nullable, got int|null. To fix this we added cast to int which should be imho fine.

Reply:
Fixed.